apps/test-quadrature64cpr/src/main.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Logger.h"
+#include "Quadrature64CPR.h"
+
+void read_all();
+
+// first pin is yellow wire
+// second pin is white wire
+constexpr Encoder::Quadrature64CPR::Config config = {PA_5, PA_6, NC, 0};
+Encoder::Quadrature64CPR enc(config);
+
+Timer theta_timer, theta_dot_timer;
+float theta     = 0;
+float theta_dot = 0;
+
+int main() {
+  while (true) {
+    theta_timer.reset();
+    theta_dot_timer.reset();
+    read_all();
+  }
+}
+
+void read_all() {
+  // angle reading
+  theta_timer.start();
+  if (enc.getAngleDeg(theta)) {
+    theta_timer.stop();
+    printf("Angle reading took: %lluus\tAngle: %.3f\r\n", theta_timer.elapsed_time().count(), theta);
+  } else {
+    // here we can log the error in the future
+    theta_timer.stop();
+    printf("Error reading angle\r\n");
+  }
+  ThisThread::sleep_for(500ms);
```

Comment:
So its still screwed up if there's no sleep here, right?

Reply:
Yea I cant call read twice in a row that's why I need to sleep.